components/net/fetch/methods.rs

```diff
@@ -1009,12 +1101,14 @@ fn should_upgrade_mixed_content_request(request: &Request) -> bool {
     }
 
     // Step 1.3
-    if !does_settings_prohibit_mixed_security_contexts(&url) {
+    if does_settings_prohibit_mixed_security_contexts(request) ==
+        MixedSecurityProhibited::NotProhibited
+    {
         return false;
     }
 
     // Step 1.4 : request’s destination is not "image", "audio", or "video".
-    if matches!(
+    if !matches!(
```

Comment:
Oh nice!

Reply:
Yeah, this one surprised me and also can be confusing to reason about with all the negation involved (do _not_ upgrade request if _not_ one of ....)